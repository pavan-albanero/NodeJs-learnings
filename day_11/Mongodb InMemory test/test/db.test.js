const { MongoMemoryServer } =require('mongodb-memory-server');
const chai = require("chai");
//let expect = chai.expect;
let chaiHttp = require("chai-http");
chai.use(chaiHttp);
const dbHandler=require("./db-handler");
let server = require('../app.js');
const logger=require('../util/logger.js');

beforeAll(async () => await dbHandler.dbConnect());

/**
 * Remove and close the db and server.
 */
afterAll(async () => await dbHandler.dbDisconnect());

describe("POST /users", () => {

    /* beforeEach(() => {
       createUser.mockReset()
       createUser.mockResolvedValue(0)
     });*/
 
     it(" get users",async ()=> 
         {
             try {
             const res = await chai.request(server).get("/users");
             
             expect(res.status).toBe(200);
             }catch(err) {
                 console.log(err);
                 logger.error("",err);
             }
         });
      } );

      describe("users",()=>{
        describe("/GET /users/:id",()=>{
            it(" get one user",async ()=> 
            {
                try {
                const res = await chai.request(server).get("/users/614b1dcb62993a8ef6f47812");
                //expect(res.statusCode).to.equal(200);
                expect(res.status).toBe(200);
                expect(res.body).toMatchSnapshot();
                }catch(err) {
                    console.log(err);
                    logger.error("",err);
                }
            });
        });
    });
   
      describe("/POST/users",()=>{
        it("should add users",async()=> 
        {
            try {
            let user={
                firstname:"pavan",
                lastname:"dulam",
                age:23,
                followers: {},
                followings: {}
            }
            const res = await chai.request(server).post("/users").send(user);
            expect(res.status).toBe(200);
            expect(res.body).toMatchSnapshot();
           
          
        }catch(err) {
            console.log(err);
            logger.error("",err);
        }
        });
    });
    describe("/PATCH/users",()=>{
        it("should update users",async()=> 
        { 
            try {
            let user={
                firstname:"sham",
                lastname:"dulam",
                age:26,
                followers: {},
                followings:{}
            }
           // let user_id=userdata;
            const res = await chai.request(server).patch(`/users/614d7e65272093c24d506sad`).send(user);
            expect(res.status).toBe(200);
            expect(res.body).toMatchSnapshot();
        }catch(err) {
            console.log(err);
            logger.error("",err);
        }
        });
    }
    )
    describe("/DELETE/users/:id",()=>{
        try {
        it("should delete users",async()=> 
        {
           // let user_id=userdata;
            const res = await chai.request(server).delete(`/users/614d7e65272093c24d506ddd`);
            expect(res.status).toBe(200);
            
        })
    }catch(err) {
        console.log(err);
        logger.error("",err);
    }
    });

    

    describe("/GET/users/:id/followers",()=>{
        try {
        it("should show followers",async()=> 
        {
            //let user_id=userdata;
            //let followUser='614b1dcb62993a8ef6f47812';
            const follower = await chai.request(server).get(`/users/614d7e65272093c24d506ddb/followers`)
            
            //console.log("followers=",follower.body);
         
            expect(follower.status).toBe(200);
          
            
        })
    }catch(err) {
        console.log(err);
        logger.error("",err);
    }
    });
