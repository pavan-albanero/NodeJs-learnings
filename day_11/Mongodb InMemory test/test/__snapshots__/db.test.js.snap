// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/PATCH/users should update users 1`] = `"update user API failed"`;

exports[`/POST/users should add users 1`] = `
Object {
  "acknowledged": true,
  "insertedId": "6152cd1016b24aa5b6cf37bf",
}
`;

exports[`/POST/users should get users 1`] = `
Object {
  "acknowledged": true,
  "insertedId": "6152cacbb032fad15cad319c",
}
`;

exports[`users /GET /users/:id  get one user 1`] = `
Object {
  "_id": "614b1dcb62993a8ef6f47812",
  "email": "pratikdulam@gmail.com",
  "firstname": "pratik",
  "followers": Array [],
  "followings": Array [
    Object {
      "_id": "614b1dbf62993a8ef6f47811",
    },
    Object {
      "_id": "614c0713f01bd3c9e913d7ee",
    },
    Object {
      "_id": "614d7e65272093c24d506ddb",
    },
    Object {
      "_id": "614d7e65272093c24d506ddb",
    },
  ],
  "lastname": "dulam",
}
`;
